protos/flyteidl/plugins/dask.proto

```diff
@@ -0,0 +1,41 @@
+syntax = "proto3";
+
+import "flyteidl/core/tasks.proto";
+
+package flyteidl.plugins;
+
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/plugins";
+
+
+// Custom Proto for Dask Plugin.
+message DaskJob {
+    // Spec for the scheduler pod.
+    Scheduler scheduler = 1; 
+
+    // Spec of the default worker group.
+    WorkerGroup workers = 2;
+}
+
+// Specification for the scheduler pod.
+message Scheduler {
+    // Optional image to use. If unset, will use the default image.
+    string image = 1;
+
+    // Resources assigned to the scheduler pod.
+    core.Resources resources = 2;
+}
+
+message WorkerGroup {
+    // Number of workers in the group.
+    uint32 number_of_workers = 1;
+
+    // Optional image to use for the pods of the worker group. If unset, will use the default image.
+    string image = 2;
+    
+    // Resources assigned to the all pods of the worker group.
+    // As per https://kubernetes.dask.org/en/latest/kubecluster.html?highlight=limit#best-practices 
+    // it is advised to only set limits. If requests are not explicitly set, the plugin will make
+    // sure to set requests==limits.
+    // The plugin sets ` --memory-limit` as well as `--nthreads` for the workers according to the limit.
+    core.Resources resources = 3;
+}
```

Comment:
Can we either start prefixing all messages with the name of the plugin or create separate go packages for them?
In python, these will be created under dask_pb2.Scheduler which is fine but in go, it'll be just plugins.Scheduler which isn't very distinguishable ...

Reply:
@hamersaw @eapolinario

Reply:
Great point, thanks! I would vote for just prefixing each with `Dask`, but could probably be convinced either way.

Reply:
Thanks @EngHabu! I've prefixed everything with `Dask` in https://github.com/flyteorg/flyteidl/pull/339/commits/6ae78d676b6677b9f66de6e62b230056a920bfd9 - let me know if you'd rather have a distinct package